src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java

```diff
@@ -85,8 +68,51 @@ public String getInventoryName() {
 	}
 
 	@Override
+	public void widgetActionPerformed(WidgetBase widget) {
+		if (widget instanceof WidgetButtonFluidicCompressor) {
+			fluidicCompressor.fill = ((WidgetButtonFluidicCompressor) widget).fill;
+			PacketHandler.instance.sendToServer(new MessageFluidicCompressorA(fluidicCompressor.fill, fluidicCompressor));
+		}
+	}
+
+	@Override
 	public void initialize() {
-		addWidget(new WidgetFluidTank(0, (width - xSize) / 2 + 53, (height - ySize) / 2 + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetFluidTank(0, guiLeft + 53, guiTop + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetButtonFluidicCompressor(1, guiLeft + 20, guiTop + 34, 19, 16, this, fluidicCompressor.fill));
+	}
+
+	public static class WidgetButtonFluidicCompressor extends WidgetBase {
+
+		private static final ResourceLocation fillMode = new ResourceLocation("bcadditions", "textures/gui/Pieces/fillMode.png");
+		private static final ResourceLocation emptyMode = new ResourceLocation("bcadditions", "textures/gui/Pieces/emptyMode.png");
+
+		private boolean fill;
+
+		public WidgetButtonFluidicCompressor(int id, int x, int y, int width, int height, GuiBase gui, boolean fill) {
+			super(id, x, y, width, height, gui);
+			this.fill = fill;
+		}
+
+		@Override
+		public void render(int mouseX, int mouseY) {
+			if (enabled) {
+				if (fill)
+					RenderUtils.drawImage(fillMode, x, y, width, height);
+				else
+					RenderUtils.drawImage(emptyMode, x, y, width, height);
+			}
+		}
+
+		@Override
+		public void onWidgetClicked(int x, int y, int button) {
+			fill = !fill;
```

Comment:
this logic might work but is not really logical, it should only swap when you press the light gray button